### PR TITLE
refactor!(ui, llc): Attachments refactor.

### LINKED ISSUE #1545: `urlAttachmentBackgroundColor` applies to whole message instead of just url attachment part
### Which packages are you using?

stream_chat_flutter

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

stream_chat_flutter - 6.1.0

### What happened?

When `urlAttachmentBackgroundColor` is set it changes the whole color of the message widget instead of just the url attachment part in the message.

Instead it should just change the color of the url attachment part's background color with a margin on all sides, so it will feel like the part/inside of the whole message. (like in WhatsApp)

https://github.com/GetStream/stream-chat-flutter/assets/74326345/17e26e38-7ecc-49f1-81e6-ce17e6570248

### Steps to reproduce

```bash
1. Create new flutter app
2. Copy below sample code in lib/main.dart
3. Add `stream_chat_flutter` dependency
4. Run code and paste the below sample message in the input field
5. Send 2 messages, one with preview and one without preview.
6. See one with preview has fully different color from normal message
```

### Supporting info to reproduce

<details><summary>Code Sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:stream_chat_flutter/stream_chat_flutter.dart';

Future<void> main() async {
  final client = StreamChatClient(
    's2dxdhpxd94g',
    logLevel: Level.INFO,
  );

  await client.connectUser(
    User(id: 'super-band-9'),
    '''eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoic3VwZXItYmFuZC05In0.0L6lGoeLwkz0aZRUcpZKsvaXtNEDHBcezVTZ0oPq40A''',
  );

  final channel = client.channel('messaging', id: 'godevs');

  channel.watch();

  runApp(
    MyApp(
      client: client,
      channel: channel,
    ),
  );
}

class MyApp extends StatelessWidget {
  const MyApp({
    super.key,
    required this.client,
    required this.channel,
  });

  final StreamChatClient client;

  final Channel channel;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      builder: (context, widget) {
        return StreamChat(
          client: client,
          streamChatThemeData: StreamChatThemeData(
            ownMessageTheme: StreamMessageThemeData(
              messageBackgroundColor: Colors.greenAccent,
              reactionsBorderColor: Theme.of(context).colorScheme.primary,
              urlAttachmentBackgroundColor: Colors.blue,
              urlAttachmentTitleMaxLine: 2,
            ),
          ),
          child: widget,
        );
      },
      home: StreamChannel(
        channel: channel,
        child: const ChannelPage(),
      ),
    );
  }
}

class ChannelPage extends StatelessWidget {
  const ChannelPage({
    super.key,
  });

  @override
  // ignore: prefer_expression_function_bodies
  Widget build(BuildContext context) {
    return const Scaffold(
      appBar: StreamChannelHeader(),
      body: Column(
        children: <Widget>[
          Expanded(
            child: StreamMessageListView(),
          ),
          StreamMessageInput(),
        ],
      ),
    );
  }
}

``` 
</details> 

<details><summary>Sample Message</summary>

```txt
Hello guys,
This is a normal message with a link at the end.

You can see the color of the message widget with with link and without link.

https://YouTube.com
```
</details> 

### Relevant log output

_No response_

### Flutter analyze output

```shell
Analyzing 5amClub-Client-App...                                         
No issues found! (ran in 4.2s)
```

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 3.10.0, on macOS 13.3.1 22E772610a darwin-arm64, locale en-IN)
    • Flutter version 3.10.0 on channel stable at /Users/omphatak/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (5 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/omphatak/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.78.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.64.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 967F1AF5-16F6-459D-8ACC-CEBD837E54B9 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.3.1 22E772610a
      darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome
      113.0.5672.92

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1621: Control Background Colors Of Attachments or No boarder Option
### Please select which package this feature is related to

stream_chat_flutter

### Which platforms would this feature impact?

iOS, Android, Web, Windows, MacOS, Linux

### Is your feature request related to a problem?

Right now all attachments share the same background colors

### Describe the solution that you'd like

To add an attachment background color option to the StreamChatThemeData for ownMessageTheme and otherMessageTheme.

### Describe alternatives that you have considered

I tried overriding the messageWidget for when it is an attachment but that kicks errors for font sizes and leaves the attachments blank if a don't want a background.

### Additional context

IMG_0026

The image above shows how a custom attachment looks with a background and that is not the look we are going for

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by xsahil03x**
Fixes: #1621, #1545

|                                                           **Before**                                                           |                                                           **After**                                                            |
|:------------------------------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------:|
| <video src='https://github.com/GetStream/stream-chat-flutter/assets/25670178/a82cd23f-a7a4-40c1-85f1-9c473e615d2e' width=180/> | <video src='https://github.com/GetStream/stream-chat-flutter/assets/25670178/22ab20ab-df86-4981-9755-07c84b681a01' width=180/> |

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart [hunk 1]
```diff
@@ -100,185 +99,157 @@ class StreamMessageWidget extends StatefulWidget {
...
+  //     final color = StreamChatTheme.of(context).colorTheme.borders;
+  //     final border = RoundedRectangleBorder(
+  //       side: attachmentBorderSide ?? BorderSide(color: color),
+  //       borderRadius: attachmentBorderRadiusGeometry ?? BorderRadius.zero,
+  //     );
+  //
+  //     if (attachments.length > 1) {
+  //       return WrapAttachmentWidget(
+  //         attachmentShape: border,
+  //         attachmentWidget: Material(
```
Comment (kanat) [inline packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart:116]: Do we still need this commented out section?

File: packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart [hunk 2]
```diff
@@ -41,18 +42,16 @@ enum DisplayWidget {
...
     required this.messageTheme,
     this.reverse = false,
     this.translateUserAvatar = true,
     this.shape,
-    this.attachmentShape,
     this.borderSide,
-    this.attachmentBorderSide,
     this.borderRadiusGeometry,
-    this.attachmentBorderRadiusGeometry,
+    this.attachmentShape,
```
Comment (kanat) [inline packages/stream_chat_flutter/lib/src/message_widget/message_widget.dart:54]: Possible breaking change?

File: packages/stream_chat_flutter/lib/src/video/video_thumbnail_image.dart [hunk 3]
```diff
@@ -1,162 +1,149 @@
...
-    this.errorBuilder,
-    this.placeholderBuilder,
+    required this.video,
+    this.headers,
+    this.imageFormat = ImageFormat.PNG,
+    this.maxHeight = 0,
+    this.maxWidth = 0,
+    this.timeMs = 0,
+    this.quality = 10,
+    this.scale = 1.0,
```
Comment (kanat) [inline packages/stream_chat_flutter/lib/src/video/video_thumbnail_image.dart:55]: Seems like a breaking change as well.

**Comment by xsahil03x** [comment]
@kanat yes, this PR will be part of V7 of the SDK. We need to write migration guides and docs along with it.

File: packages/stream_chat/lib/src/core/models/attachment_giphy_info.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  const GiphyInfoType(this.value);
+
+  /// The value of the [GiphyInfoType].
+  final String value;
+}
+
+/// {@template giphy_info}
+/// A class that contains extra information about a Giphy attachment.
+/// {@endtemplate}
+class GiphyInfo {
```
[the remark this answers is not in the document]
Comment (xsahil03x) [reply packages/stream_chat/lib/src/core/models/attachment_giphy_info.dart:32]: not needed. Used just to store some values.

**Review by kanat — APPROVED** [review]
LGTM

**Pull request merged**